### ISSUE TITLE
[Python] CWE-943: Add NoSQL Injection Query

### ISSUE THREAD

**Issue by mrthankyou** [github/securitylab#342]
## Query
Relevant PR: https://github.com/github/codeql/pull/5612

## CVE ID(s)

*List the CVE ID(s) associated with this vulnerability. GitHub will automatically link CVE IDs to the [GitHub Advisory Database](https://github.com/advisories).*

- WIP

## Report

This query queries for NoSQL injections. NoSQL injections leave applications vulnerable to data exportation, data manipulation, authorization bypass, and more. Specifically this query looks for vulnerabilities that utilize any one of these three major Python MongoDB libraries:

- [PyMonogo](https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html)
- [Flask-PyMongo](https://flask-pymongo.readthedocs.io/en/latest/)
- [MongoEngine](https://github.com/MongoEngine/mongoengine)
- [Flask MongoEngine](https://github.com/MongoEngine/flask-mongoengine)

The most common scenario this query looks for is for a Python application to accept user input from a HTTP request, convert that HTTP request to a Python object, and then inject that object directly into a sink method provided by one of the three libraries above.

- [X] Are you planning to discuss this vulnerability submission publicly? (Blog Post, social networks, etc). *We would love to have you spread the word about the good work you are doing*

## Result(s)

- [STATUS: contacting developer](#)

**Comment by mrthankyou** [comment]
I've written an article about this query and posted the link below. I'm working on getting a blog set up to post this. In the meantime I've created a Git Gist.

https://gist.github.com/mrthankyou/5bfc5c06de2a7b2d0a1da8425dc0baba

**Comment by ghsecuritylab** [comment]
Your submission is now in status **SecLab review**.

_For information, the evaluation workflow is the following:
SecLab review > FP Check > CodeQL review > SecLab finalize > Pay > Closed_

**Comment by antonio-morales** [comment]
Hi @mrthankyou !

You need to provide a finding in order to meet the submission criteria.

Can you provide a valid result for this query?

**Comment by mrthankyou** [comment]
@antonio-morales,

Yes I can. I haven't done so on here since I've tried contacting numerous repos and have yet to receive a response from the repo owners. In an attempt to protect any repo that has this vulnerability from being ousted before fixing this issue (or at the very least have a PR up for them to merge) how should I proceed here?

**Comment by anticomputer** [comment]
@mrthankyou we'll move this forward based on results provided in private to the CodeQL team, no need to disclose unfixed vulnerabilities.

**Comment by ghsecuritylab** [comment]
Your submission is now in status **FP Check**.

_For information, the evaluation workflow is the following:
SecLab review > FP Check > CodeQL review > SecLab finalize > Pay > Closed_

**Comment by mrthankyou** [comment]
Thanks @anticomputer!

**Comment by ghsecuritylab** [comment]
Your submission is now in status **Generate Query Results**.

_For information, the evaluation workflow is the following:
SecLab review > Generate Query Results > FP Check > CodeQL review > SecLab finalize > Pay > Closed_

**Comment by ghsecuritylab** [comment]
Your submission is now in status **FP Check**.

_For information, the evaluation workflow is the following:
SecLab review > Generate Query Results > FP Check > CodeQL review > SecLab finalize > Pay > Closed_

**Comment by ghsecuritylab** [comment]
Your submission is now in status **CodeQL review**.

_For information, the evaluation workflow is the following:
SecLab review > Generate Query Results > FP Check > CodeQL review > SecLab finalize > Pay > Closed_

**Comment by ghsecuritylab** [comment]
Your submission is now in status **SecLab finalize**.

_For information, the evaluation workflow is the following:
SecLab review > Generate Query Results > FP Check > CodeQL review > SecLab finalize > Pay > Closed_
